packages/canary-web-components/src/utils/helpers.ts

```diff
@@ -664,15 +664,18 @@ export const addDataToPosition = (
 ): IcDataTableDataType => {
   const newData: IcDataTableDataType = {};
   const newIndexes = newKeys.map((key) => key.index);
-  let controlledIndex = 0; // When a new key is added to the data, need to increment the index to account for this new object value
 
-  Object.keys(dataObject).forEach((dataKey) => {
-    if (newIndexes.includes(controlledIndex)) {
-      newData[newKeys[newIndexes.indexOf(controlledIndex)].key] = newValue;
-      controlledIndex++;
+  const keys = Object.keys(dataObject);
```

Comment:
You should use Object.entries to get the keys and values from one expression. Use destructuring to assign them to the key, value variables. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries

Reply:
I tried that and wasn't able to get it to work, can you provide a GitHub suggestion?

Reply:
You'll need to use map to normalize the array. But that's more work than it should be so this is ok